### PR TITLE
docs: ADR Athens System Design.

### PR DESCRIPTION
**Pull request by neotyk**

### CODE REVIEW

File: doc/adr/0017-system-design-lan-party.dot [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        cSE -> "Optimistic Update" [label="Apply optimistically";];
+        cSE -> SendIt [label="Send AGO over WS";];
+        "Optimistic Update" -> cResolution [label="Composed of AGO";];
+        cResolution -> cTransact;
+        cTransact -> cDatascript;
+
+        cAck -> cDatascript [label="Updates DB, cleaning Optimistic state";];
+        cDatascript -> UI [style=dotted; label="Render based on DB context"; color=blue];
+        cDatascript -> cResolution [style=dotted; label="Resolved in DB context"; color=blue];
+        cDatascript -> cSE [style=dotted; label="Creates AGO in DB context"; color=blue];
```
Comment (filipesilva) [inline doc/adr/0017-system-design-lan-party.dot:26]: I don't understand what this arrow means. Can you explain?

File: doc/adr/0017-system-design-lan-party.dot [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        sPresence [label="Forward to all connected clients";];
+        sForward [label="Forward AGO to all connected clients";];
+        sEventStore [label="Event Store";];
+        
+        wsReceive -> isAGO;
+        isAGO -> sAGO [label="Yes, AGO";];
+        sAGO -> sResolution [label="1";];
+        sAGO -> sAck [label="2";];
+        sAGO -> sForward [label="3";];
+        sAGO -> sEventStore [label="4";];
```
Comment (filipesilva) [inline doc/adr/0017-system-design-lan-party.dot:48]: What are the numbers here meant to convey? Is it order of operations?

If it is order, then 4 is actually the very first one and blocks the others (as it should).

File: doc/adr/0017-system-design-lan-party.dot (hunk 1 above)
Comment (filipesilva) [reply doc/adr/0017-system-design-lan-party.dot:26]: Thinking a bit more about it, I think it means that semantic events create AGO using the DB. Is that right? If so, then I think all blue arrows are pointing in the wrong direction.

e.g. `cDatascript` doesn't use `AGO resolution` for `Resolved in DB context`, instead `AGO resolution` uses `cDatascript`

**Review by filipesilva — APPROVED** [review]
lgtm, except for the two comments

File: doc/adr/0017-system-design-lan-party.dot (hunk 1 above)
Comment (neotyk) [reply doc/adr/0017-system-design-lan-party.dot:26]: State of Datascript DB influences resolution.

File: doc/adr/0017-system-design-lan-party.dot (hunk 2 above)
Comment (neotyk) [reply doc/adr/0017-system-design-lan-party.dot:48]: Yes, it's order of execution, I'll update.

File: doc/adr/0017-system-design-lan-party.dot (hunk 1 above)
Comment (neotyk) [reply doc/adr/0017-system-design-lan-party.dot]: Updated

**Pull request merged**